src/kit/DataGrid/DataGrid.tsx

```diff
@@ -0,0 +1,597 @@
+  return !!(cell as LinkCell).data?.link?.href;
+};
+
+export function DataGrid<T, ContextAction = void | string, ContextActionData = void>({
+  columns,
+  data,
+  getHeaderMenuItems,
+  getRowAccentColor,
+  hideUnpinned = false,
+  height = '100%',
+  onColumnResize,
+  onContextMenuComplete,
+  onPinnedColumnsCountChange,
+  onSelectionChange,
+  onColumnsOrderChange,
+  isPaginated = false,
+  page,
+  pageSize,
+  onPageUpdate,
+  pinnedColumnsCount = 0,
+  renderContextMenuComponent,
+  rowHeight = 36,
+  selection = {
+    columns: CompactSelection.empty(),
+    rows: CompactSelection.empty(),
+  },
+  total,
+  imperativeRef,
+  sorts = [],
+  staticColumns,
+  smoothScrollX = true,
+  smoothScrollY = true,
+}: DataGridProps<T, ContextAction, ContextActionData>): JSX.Element {
+  const gridRef = useRef<DataEditorRef>(null);
+  const clickedCellRef = useRef<{ col: number; row: number } | null>(null);
+  const [hoveredRow, setHoveredRow] = useState<number>();
+  const [scrollPositionSetCount] = useState(observable(0));
+
+  const onScroll = useCallback(
+    ({ y, height }: Rectangle) => {
+      if (
+        isPaginated ||
+        _.isUndefined(pageSize) ||
+        scrollPositionSetCount.get() < SCROLL_SET_COUNT_NEEDED
+      )
+        return;
+      onPageUpdate?.(Math.floor((y + height) / pageSize));
+    },
+    [scrollPositionSetCount, onPageUpdate, pageSize, isPaginated],
+  );
+
+  useEffect(() => {
+    if (scrollPositionSetCount.get() >= SCROLL_SET_COUNT_NEEDED) return;
+    if (gridRef.current !== null && !_.isUndefined(page) && !_.isUndefined(pageSize)) {
+      const rowOffset = Math.max(page * pageSize, 0);
+      const bounds = gridRef.current.getBounds(0, rowOffset);
+      if (bounds && !Number.isNaN(bounds.x)) {
+        gridRef.current.scrollTo(0, rowOffset);
+        scrollPositionSetCount.update((x) => x + 1);
+      }
+    }
+  });
+
+  const [headerMenuIsOpen, setHeaderMenuIsOpen] = useState(false);
+  const [headerMenuProps, setHeaderMenuProps] = useState<Omit<HeaderMenuProps, 'open'>>({
+    bounds: {
+      height: 0,
+      width: 0,
+      x: 0,
+      y: 0,
+    },
+    handleClose: () => setHeaderMenuIsOpen(false),
+  });
+
+  const [contextMenuOpen] = useState(observable(false));
+  const contextMenuIsOpen = useObservable(contextMenuOpen);
+
+  const [contextMenuProps, setContextMenuProps] = useState<null | Omit<
+    ContextMenuProps<T, ContextAction, ContextActionData>,
+    'open'
+  >>(null);
+
+  const {
+    getThemeVar,
+    themeSettings: { theme: appTheme },
+  } = useTheme();
+  const theme = getTheme(appTheme);
+
+  const isMobile = useMobile();
+
+  const headerIcons = useMemo(() => getHeaderIcons(appTheme), [appTheme]);
+
+  const { tooltip, onItemHovered, closeTooltip } = useTableTooltip<T>({
+    columns,
+    data,
+  });
+
+  const scrollToTop = useCallback(() => {
+    if (gridRef.current) {
+      gridRef.current.scrollTo(0, 0);
+    }
```

Comment:
nit: Should be able to make it `gridRef.current?.scrollTo(0, 0);`. Also, refs don't have to go into the dependency array, right?